moped-editor/src/views/projects/projectView/ProjectFundingTable.js

```diff
@@ -305,25 +294,233 @@ const ProjectFundingTable = () => {
     setSnackbarState(DEFAULT_SNACKBAR_STATE);
   };
 
-  /**
-   * Column configuration for <MaterialTable>
-   */
-  const columns = [
+  const handleRowModesModelChange = (newRowModesModel) => {
+    setRowModesModel(newRowModesModel);
+  };
+
+  // adds a blank row to the table and updates the row modes model
+  const handleAddRecordClick = () => {
+    // use a random id to keep track of row in row modes model and data grid rows
+    // before the record is added to the db
+    const id = Math.floor(Math.random() * 10000);
```

Comment:
Does the ID need to be numeric? Do you think we could expand the namespace here? This is the birthday paradox, and given tables of 5 records each, I *think* a collision is expected around every 430 table loads. The most back of the napkin guess there, so please don't quote me.

Can we use a UUID? from the `uuid` package?

Reply:
good question, i didnt think about collisions. the row ids are usually the project funding id, and it could randomly duplicate one of those ids.

Reply:
added this to https://github.com/cityofaustin/atd-data-tech/issues/16831